ethereum/core/src/main/java/org/hyperledger/besu/ethereum/storage/keyvalue/WorldStateKeyValueStorage.java

```diff
@@ -156,12 +184,18 @@ public Updater putAccountStorageTrieNode(final Bytes32 nodeHash, final Bytes nod
 
     @Override
     public void commit() {
-      nodeAddedListeners.forEach(listener -> listener.onNodesAdded(addedNodes));
-      transaction.commit();
+      lock.lock();
+      try {
+        nodeAddedListeners.forEach(listener -> listener.onNodesAdded(addedNodes));
+        transaction.commit();
+      } finally {
+        lock.unlock();
+      }
```

Comment:
Not so fast :) I'm going to start some fast syncs to see what it does with the full state because a longer sweeping phase means a larger storage spike caused by the snapshot of the column while we're iterating.

On the other hand, keeping up with head means rare commits to the world state database so a lot less contention there.

Reply:
Fortunately it's the mark phase that takes the vast majority of the time - the sweep phase IIRC was quite fast (it may not be so now).  Might still be an issue but it's still a much lower impact than I was first expecting when we started talking about needing a lock.